### DISCUSSION TITLE
What does a modern command line look like for sqlcmd/bcp?

### DISCUSSION THREAD

**Discussion by shueybubbles** [microsoft/go-sqlcmd#113] [General]
The current `sqlcmd` implementation in this repo is designed to be a dropin replacement for the ODBC-based version as much as possible. Going forward we want to create a more modern command that has a richer set of verbs much like `az` and `kubectl` offer a breadth of functionality for Azure and Kubernetes respectively. 

As we start an implementation of `bcp` that has a similar goal to replace the existing `bcp`, we plan to create a new executable that offers both `bcp` and `sqlcmd` functionality, distinguished by verbs passed on the command line.

We are looking for input on what this new parameter set should look like to make it readily adaptable into your CI/CD pipelines or administrative workflows.

Some questions to ponder:
1. What should the name of the executable be? 
2. What should the verbs be? `query` is likely a good verb for the `sqlcmd` functionality, while `bulkcopy` could work for `bcp`. 
3. Are there switches from `sqlcmd` and `bcp` that we could skip porting over to the combined tool, as not being required for modern tools? 
4. Besides query and bulkcopy functionality, what other SQL Server management capabilities are desirable in a cross platform CLI?  We've thought about integrations with Docker and DacFx to provide a one-stop-shopping source for instantiating a SQL Server instance, deploying a database to it, and querying it.

**Comment by match-gabeflores** [comment]
somewhat related - but it would be fantastic if sqlcmd (or even a diff msft tool) could be used to bulk copy **parquet** files (import or export). The current approaches for loading parquet files to on-prem mssql aren't especially quick. Most modern data movement relies on parquet instead of csv

edit:
https://github.com/microsoft/go-sqlcmd/issues/486
